src/authorization.js

```diff
@@ -258,6 +268,12 @@ class Authorization {
     return this.accessModes[ Authorization.acl.CONTROL ]
   }
 
+  clone () {
+    let auth = new Authorization()
+    Object.assign(auth, JSON.parse(JSON.stringify(this)))
```

Comment:
I understand your concern - normally `Object.assign()` is vulnerable when you mutate the original object. However, the `JSON.parse(JSON.stringify())` usage fixes that -- the serializing and re-parsing breaks the ties/references to the original object. It is, essentially, a deep copy operation.
I'm using this approach for two reasons - 1) it avoids an additional dependency on lodash, and 2) from what I understand, this performs even faster on modern js engines than cloneDeep.